numpy/f2py/tests/src/isocintrin/isoCtests.f90

```diff
@@ -0,0 +1,17 @@
+  module coddity
+    use iso_c_binding, only: c_double, c_int
+    implicit none
+    contains
+      subroutine c_add(a, b, c) bind(c, name="c_add")
+        real(c_double), intent(in) :: a, b
+        real(c_double), intent(out) :: c
+        c = a + b
+      end subroutine c_add
+      ! gh-9693
+      function wat(x, y) result(z) bind(c, name='wat')
+          integer(c_int), intent(in) :: x, y
+          integer(c_int) :: z
+
+          z = x + 7
+      end function wat
```

Comment:
Could you add a test with no name kwarg? If it is not valid fortran, it should raise.

Reply:
It is [valid still](https://www.intel.com/content/www/us/en/docs/fortran-compiler/developer-guide-reference/2023-0/bind-c.html) (just defaults to binding the same name in Fortran and C), but its still good to test it, I modified this to be without `name`.

`function wat(x) result (z) bind(c)` --> same Fortran name in C
`function wat(x) result (z) bind(c, name="blah") --> `wat` in Fortran, `blah` in C